### ISSUE TITLE
GNS-TAT implementations

### ISSUE THREAD

**Issue by newalexander** [shehzaidi/pre-training-via-denoising#9]
Thank you for making this compelling work available. I am interested in using the GNS-TAT model implementation you refer to in your paper. Would it be possible for you to clarify which of the model architectures in this repo corresponds to the GNS-TAT model? Thank you!

**Comment by newalexander** [comment]
I would also be interested, if possible, in further details on the process of pretraining PCQM4Mv2. In your paper, you say that you ran pretraining for `3 · 10^5` training steps. Did the pretraining loss decrease for all of this training? Did you experiment with finetuning from earlier checkpoints? Thank you!

**Comment by shehzaidi** [comment]
Hi @newalexander, thanks for your questions!

Regarding GNS-TAT, unfortunately we haven't open-sourced the code for that, so it is not available in this repo (only the Torch-MDNet architecture is available in this repo). GNS-TAT is constructed starting from GNS (an implementation of which is available [here](https://github.com/google-deepmind/deepmind-research/blob/master/learning_to_simulate/graph_network.py)) using the modifications we outlined in the [paper](https://arxiv.org/abs/2206.00133) in Appendix A. Sorry about not having a readily usable open source implementation of GNS-TAT!

Regarding the pre-training steps, that's a great question. I previously tried reducing the number of pre-training steps to 200K steps and didn't notice any significant degradation in fine-tuning performance. The training loss decreases quickly initially and then slowly reduces in the latter half of training. Here's a plot of the training loss (the different colors are different sized models, the darker being the largest):

<img width="436" alt="image" src="https://github.com/shehzaidi/pre-training-via-denoising/assets/19708290/eb08e63f-a79f-4b3a-b873-bc1144777112">

It's possible one could get away with pre-training for even fewer steps (before the train loss converges), but I haven't tried that!
